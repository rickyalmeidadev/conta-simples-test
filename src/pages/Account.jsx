import React, { useState, useEffect } from 'react';
import { BounceLoader } from 'react-spinners';
import { getUser } from '../services/services';
import Navbar from '../components/Navbar';
import AccountCard from '../components/AccountCard';
import cards from '../utils/cards.json';

const Account = () => {
  const [user, setUser] = useState(null);

  useEffect(() => {
    getUser()
      .then((response) => {
        setUser(response.data);
      })
      .catch(console.error);
  }, []);

  if (!user) {
    return (
      <div className="loading">
        <BounceLoader color="#7be115" size="5rem" />
      </div>
    );
  }

  return (
    <div className="wrapper">
      <Navbar />
      <header className="container">
        <h2 className="title title--green">Minha conta</h2>
      </header>
      <div className="container">
        <p className="info">
          <span className="info__text">Saldo atual: </span>
          <span className="info__value">R$ {user.balance.toFixed(2)}</span>
        </p>
      </div>

      <section className="container account-grid">
        {cards.map((card) => (
          <AccountCard key={card.test} {...card} />
        ))}
      </section>
    </div>
  );
};

export default Account;
