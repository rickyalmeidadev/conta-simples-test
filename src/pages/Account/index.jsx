import React, { useState, useEffect } from 'react';
import { getUser } from '../../services/services';

const Account = () => {
  const [user, setUser] = useState(null);

  useEffect(() => {
    getUser().then(response => {
      setUser(response.data);
    }).catch(console.error);
  }, []);

  return (
    <>
      <h1>Bem-vindo(a), Nome</h1>

      <section>
        <article data-testid="statement">
          <h2>Estrato</h2>
          <p>Lorem ipsum dolor sit amet.</p>
          <button type="button">Ver estrato</button>
        </article>
      </section>
    </>
  );
};

export default Account;
