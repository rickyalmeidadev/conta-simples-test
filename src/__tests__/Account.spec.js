import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Account from '../pages/Account';

afterEach(cleanup);

const user = { name: 'Fernando' };

describe('<Account />', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<Account />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('shows an welcoming message with the user name', async () => {
    const mock = jest.fn(() => Promise.resolve(user));
    const { getByText } = render(<Account />);

    const response = await mock();

    expect(getByText(/bem-vindo/i)).toContains(`${response.name}`);
  });
});
