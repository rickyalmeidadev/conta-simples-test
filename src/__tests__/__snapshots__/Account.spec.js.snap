// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Account /> renders correctly 1`] = `
<DocumentFragment>
  <h1>
    Bem-vindo(a), Nome
  </h1>
  <section>
    <article
      data-testid="statement"
    >
      <h2>
        Estrato
      </h2>
      <p>
        Lorem ipsum dolor sit amet.
      </p>
      <button
        type="button"
      >
        Ver estrato
      </button>
    </article>
  </section>
</DocumentFragment>
`;
